File: algorithms/sort/quick_sort/quick_sort.rb
def quick_sort(array)
  return array if array.length < 2 # base case -> array w/ 1 or 0 items
  
  pivot = array.pop
  left = []
  right = []

  array.each do |item|
    item < pivot ? left << item : right << item
  end

  #recursive case
  quick_sort(left) + [pivot] + quick_sort(right)

end

p quick_sort([10,5,2,3])

File: algorithms/sort/quick_sort/quickSort.js
function quickSort(array) {

  if (array.length < 2) {
    return array
  }

  var pivot = array.pop
  var left = []
  var right = []

  for(var i = 0; i < array.length; i++) {
    if(array[i] < pivot) {
      left.push(array[i])
    } else {
      right.push(array[i])
    }
  }

  return quickSort(left).concat([pivot]).concat(quickSort(right))

}

quickSort([10,5,2,3])